File: ecommerce-frontend/src/services/cartService.js
import axios from "axios";

const API = "http://localhost:8080/api/cart";

// Get JWT Token
const getToken = () => {

    return localStorage.getItem("token");

};

// Authorization Header
const authHeader = () => {

    return {

        headers: {

            Authorization: `Bearer ${getToken()}`

        }

    };

};

// ==============================
// GET CART
// ==============================

const getCart = () => {

    return axios.get(

        API,

        authHeader()

    );

};

// ==============================
// ADD TO CART
// ==============================

const addToCart = (productId, quantity = 1) => {

    return axios.post(

        `${API}/add`,

        {

            productId,

            quantity

        },

        authHeader()

    );

};

// ==============================
// UPDATE CART ITEM
// ==============================

const updateQuantity = (

    cartItemId,

    quantity

) => {

    return axios.put(

        `${API}/update/${cartItemId}`,

        {

            quantity

        },

        authHeader()

    );

};

// ==============================
// REMOVE ITEM
// ==============================

const removeItem = (cartItemId) => {

    return axios.delete(

        `${API}/remove/${cartItemId}`,

        authHeader()

    );

};

// ==============================
// CLEAR CART
// ==============================

const clearCart = () => {

    return axios.delete(

        `${API}/clear`,

        authHeader()

    );

};

const cartService = {

    getCart,

    addToCart,

    updateQuantity,

    removeItem,

    clearCart

};

export default cartService;

File: ecommerce-backend/src/main/java/com/ecommerce/service/CartService.java
package com.ecommerce.service;

import com.ecommerce.dto.AddToCartRequest;
import com.ecommerce.dto.CartResponse;
import com.ecommerce.dto.UpdateCartRequest;

public interface CartService {

    // ===============================
    // Add Product To Cart
    // ===============================
    CartResponse addToCart(AddToCartRequest request);

    // ===============================
    // Get Logged In User Cart
    // ===============================
    CartResponse getCart();

    // ===============================
    // Update Quantity
    // ===============================
    CartResponse updateCartItem(
            Long cartItemId,
            UpdateCartRequest request
    );

    // ===============================
    // Remove Item From Cart
    // ===============================
    String removeCartItem(Long cartItemId);

    // ===============================
    // Clear Complete Cart
    // ===============================
    String clearCart();

}